src/app/registration/registration.component.ts

```diff
@@ -0,0 +1,63 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-registration',
+  templateUrl: './registration.component.html',
+  styleUrls: ['./registration.component.scss']
+})
+export class RegistrationComponent implements OnInit {
+
+  public firstName: string;
+  public lastName: string;
+  public email: string;
+  public password: string;
+  public passwordConfirmation: string;
+
+  private passwordRegex = new RegExp(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]{8,}$/);
+  // tslint:disable-next-line: max-line-length
+  private emailRegex = new RegExp(/^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/);
+
+  constructor() { }
+
+  ngOnInit(): void {
+  }
+
+  public sendRegistrationRequest(): void {
+    if (!this.errorsInForm()) {
+
+    }
+  }
+
+  private errorsInForm(): boolean {
+    // TODO: visually show errors
+    let errors = false;
+    if (this.firstName == null) {
```

Comment:
sostituirei (anche per il codice precedente) ogni check dove c'e' `null`.
perche' x != null, lo puoi sostituire con !!x

`this.firstName == null` non e' detto che soddisfi la condizione che stai cercando
perche' se `this.firstName = ''` allora `this.firstName == null` e' falso, mentre `!this.firstName` controlla se e' null  o se e' `''`

![image](https://user-images.githubusercontent.com/519778/92602242-31165580-f2ae-11ea-8fd2-2535a3cecf38.png)